slapo/initialization.py

```diff
@@ -59,7 +59,7 @@ def init_on_device(device: torch.device, include_buffers: bool = False):
 
     def register_empty_parameter(module, name, param):
         old_register_parameter(module, name, param)
-        if param is not None:
+        if param is not None and param.device != device:
```

Comment:
what is this used for? I think register_empty_parameter is recalled after we exit the init context?

Reply:
Without this, wte.weight = lm_head.weight won't let them have the same parameter object.

Reply:
ok so this is triggered during init?

Reply:
No it's triggered by `__setattr__` -> `register_parameter`